src/lib.rs

```diff
@@ -638,6 +638,12 @@ pub use zerocopy_derive::KnownLayout;
     not(feature = "derive"),
     doc = concat!("[derive]: https://docs.rs/zerocopy/", env!("CARGO_PKG_VERSION"), "/zerocopy/derive.KnownLayout.html"),
 )]
+#[cfg_attr(
+    zerocopy_diagnostic_on_unimplemented,
+    diagnostic::on_unimplemented(
+        note = "Consider deriving `KnownLayout` for `{Self}` (if it is defined in this crate)"
```

Comment:
In the interest of keeping things short and sweet, cut the parenthetical. Also, since `{Self}` can be arbitrarily long, it's best to end the message with the type.
```suggestion
        note = "Consider adding `#[derive(KnownLayout)]` for `{Self}`"
```

And finally, even if it's _not_ defined in the current crate: They should totally still go and file a PR on whatever crate defines `Self`. ;-)

Reply:
My concern is that this message can trigger on built-in types like `fn()` or `&T`, where that message is misleading. Or is your feeling that it'd be obvious that it shouldn't be taken literally in all circustances?

Reply:
I think it's obvious it should be taken literally, and we already soften the language by leading with "Consider".

Reply:
Done.